libs/pika/async_mpi/include/pika/async_mpi/dispatch_mpi.hpp

```diff
@@ -95,7 +97,7 @@ namespace pika::mpi::experimental::detail {
 
                 template <typename Error>
                 friend constexpr void
-                tag_invoke(ex::set_error_t, dispatch_mpi_receiver r, Error&& error) noexcept
+                tag_invoke(ex::set_error_t, dispatch_mpi_receiver&& r, Error&& error) noexcept
```

Comment:
Just to check, was the `&&` required to get this to compile? The receiver should be (cheaply) move-constructible, so both should work.

Reply:
no, it was fine but I changed that before finding the correct fix later ....

Reply:
Thanks, do you mind removing the `&&`. We've generally tried to move to taking the receivers by value since it's much simpler to reason about (and not any more expensive assuming the receiver just holds a pointer to the operation state).